cmake/kokkos_compiler_id.cmake

```diff
@@ -163,7 +163,7 @@ if(CMAKE_CXX_STANDARD EQUAL 17)
   set(KOKKOS_CLANG_CUDA_MINIMUM 10.0.0)
   set(KOKKOS_CLANG_OPENMPTARGET_MINIMUM 15.0.0)
   set(KOKKOS_GCC_MINIMUM 8.2.0)
-  set(KOKKOS_INTEL_MINIMUM 19.0.5)
+  set(KOKKOS_INTEL_MINIMUM "not supported")
```

Comment:
It would be good to treat ICC and XL uniformly since we removed support for both of them.